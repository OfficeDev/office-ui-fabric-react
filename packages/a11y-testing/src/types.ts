export type PropValue = string | number | boolean | any;
export type Props = { [name: string]: PropValue };
export type AccessibilityBehavior = (props: Props) => any;

export interface Rule {
  stringify: () => string;
}

export interface Slot {
  name: string;
  props: Props[];
  expectAttribute: boolean;
  expectedAttribute: string;
  expectedValue: PropValue;
  description: string;
  hidden: boolean;
  afterEvent: string;
  afterEventData: any[];
  checkEvent: boolean;
}

export interface TestFacade {
  slotExists(slotName: string): boolean;
  attributeExists(slotName: string, attributeName: string): boolean;
  attributeHasValue(slotName: string, attributeName: string, value: PropValue): boolean;
  getAttributeValue(slotName: string, attribute: string): PropValue;
  forProps(props: Props): TestFacade;
  afterEvent(slotName: string, eventName: string, args: any[]): void;
}
