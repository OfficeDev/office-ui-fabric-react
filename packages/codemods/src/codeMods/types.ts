import { SourceFile } from 'ts-morph';

// tslint:disable-next-line: interface-name
export interface CodeModResult {
  success?: boolean;
}

// tslint:disable-next-line: interface-name
export interface Codemod<T = SourceFile> {
  /**
   * The version before which the codemod should run.
   * Must be valid semver.
   * Currently has no impact. Maybe remove?
   */
  version?: string;
  /**
   * A string to help identify the codemod.
   */
  name: string;
  /**
   * The actual function that should be run on any given file.
   * TODO, is there a possibility of codemods that would need to execute over mutiple files?
   */
  run: (file: T) => CodeModResult;
  /**
   * If not enabled, then this mod will not be conisdered to run. Only enable it once it's ready
   * to be applied in real world scenarios
   */
  enabled?: boolean;
}
