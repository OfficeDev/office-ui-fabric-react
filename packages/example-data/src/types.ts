export interface IFooProps {
  /** @deprecated */
  dep1?: string;
  /** @deprecated */
  dep2?: string;
  notDep?: string;
}
