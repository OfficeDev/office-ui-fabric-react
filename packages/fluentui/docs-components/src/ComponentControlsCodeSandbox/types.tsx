export type CodeSandboxImport = {
  module: any;
  version: string;
  required: boolean;
};

export type SourceManagerLanguage = 'js' | 'ts';
