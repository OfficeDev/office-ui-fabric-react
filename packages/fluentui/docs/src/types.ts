import { UseKnobOptions } from '@fluentui/docs-components';
import { ThemePrepared } from '@fluentui/react-northstar';
import { FluentComponentInfo, ComponentProp } from '@fluentui/react-docgen';

// !!!!!!!!!!
// Keep in sync with scripts/docs/types.ts!
// !!!!!!!!!!
export type ExampleSource = {
  js: string;
  ts: string;
};

/**
 * Component menu item shown in the site sidebar.
 * An array of menu items is saved in `componentMenu.json`.
 */
// !!!!!!!!!!
// Keep in sync with scripts/docs/types.ts!
// !!!!!!!!!!
export type ComponentMenuItem = {
  displayName: string;
  type: string;
};

export type KnobGeneratorOptions = {
  propName?: string;
  propDef: ComponentProp;
  componentInfo: FluentComponentInfo;
  theme: ThemePrepared;
};
export type KnobDefinition = UseKnobOptions<any> & { hook: Function };

export type KnobGenerator<T> = (options: KnobGeneratorOptions) => KnobDefinition;

export type KnobComponentGenerators<P> = Partial<Record<keyof P, KnobGenerator<any>>>;
