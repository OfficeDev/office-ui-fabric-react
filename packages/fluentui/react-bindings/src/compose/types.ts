import * as React from 'react';

import { COMPOSE_PROP } from './compose';

export type ComposeInputComponent<P = {}> = React.FunctionComponent<P> & { [COMPOSE_PROP]?: ComposePreparedOptions };

export type ComposeInputOptions<
  AllProps extends Record<string, any>,
  BehaviorProps extends Record<string, any>,
  StylesProps extends Record<string, any>
> = {
  className?: string;
  displayName?: string;

  mapPropsToBehavior?: (props: AllProps) => BehaviorProps;
  mapPropsToStyles?: (props: AllProps) => StylesProps;

  handledProps?: (keyof AllProps)[];
  overrideStyles?: boolean;
};

export type ComposePreparedComponent<P = {}> = React.FunctionComponent<P> & { [COMPOSE_PROP]: ComposePreparedOptions };

export type ComposePreparedOptions = {
  className: string;
  displayNames: string[];

  mapPropsToBehaviorChain: ((props: Record<string, any>) => Record<string, any>)[];
  mapPropsToStylesChain: ((props: Record<string, any>) => Record<string, any>)[];

  handledProps: (keyof Record<string, any>)[];
  overrideStyles: boolean;
};

export type ComposeResolvedOptions = {
  className: string;
  displayNames: string[];

  behaviorProps: Record<string, any>;
  stylesProps: Record<string, any>;

  handledProps: (keyof Record<string, any>)[] | undefined;
  overrideStyles: boolean;
};
