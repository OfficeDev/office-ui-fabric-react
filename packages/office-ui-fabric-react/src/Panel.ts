export * from './packages/react-surfaces/Panel';
