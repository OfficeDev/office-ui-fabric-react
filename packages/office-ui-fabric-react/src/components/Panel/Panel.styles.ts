import { IPanelStyleProps, IPanelStyles } from './Panel.types';
import { AnimationClassNames, AnimationVariables, getGlobalClassNames, ScreenWidthMinMedium } from '../../Styling';

const GlobalClassNames = {
  root: 'ms-Panel',
  main: 'ms-Panel-main',
  commands: 'ms-Panel-commands',
  contentInner: 'ms-Panel-contentInner',
  navigation: 'ms-Panel-navigation',
  closeButton: 'ms-Panel-closeButton ms-PanelAction-close',
  header: 'ms-Panel-header',
  headerText: 'ms-Panel-headerText',
  content: 'ms-Panel-content',
  footer: 'ms-Panel-footer',
  footerInner: 'ms-Panel-footerInner'
};

export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
  const { className, isAnimating, isOpen, theme } = props;
  const { palette } = theme;

  // widthXS: '272px';
  const widthSM = '340px';
  // widthMD: '643px';
  // widthLG: '940px';
  // marginMD: '48px';
  // marginLG: '428px';
  // marginXL: '176px';
  // commandBarHeight: '44px';

  const classNames = getGlobalClassNames(GlobalClassNames, theme);

  return {
    root: [
      classNames.root,
      {
        pointerEvents: 'none',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0
      },
      className
    ],
    overlay: [
      {
        pointerEvents: 'none',
        opacity: 1,
        cursor: 'pointer',
        transition: `opacity ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
      },
      isOpen && isAnimating && AnimationClassNames.fadeIn200,
      !isOpen && isAnimating && AnimationClassNames.fadeOut200
    ],
    hiddenPanel: {
      visibility: 'hidden'
    },
    main: [
      classNames.main,
      {
        backgroundColor: palette.white,
        position: 'absolute',
        right: 0,
        width: '100%',
        bottom: 0,
        top: 0,
        overflowX: 'hidden',
        overflowY: 'auto',
        WebkitOverflowScrolling: 'touch',
        selectors: {
          ['@media (min-width: ' + ScreenWidthMinMedium + ')']: {
            borderLeft: `1px solid ${palette.neutralLight}`,
            borderRight: `1px solid ${palette.neutralLight}`,
            pointerEvents: 'auto',
            width: widthSM,
            boxShadow: '-30px, 0px, 30px, -30px, .2',
            left: 'auto'
          }
        }
      }
    ],
    commands: [classNames.commands],
    contentInner: [classNames.contentInner],
    navigation: [classNames.navigation],
    closeButton: [classNames.closeButton],
    header: [classNames.header],
    headerText: [classNames.headerText],
    content: [classNames.content],
    footer: [classNames.footer],
    footerInner: [classNames.footerInner]
  };
};
