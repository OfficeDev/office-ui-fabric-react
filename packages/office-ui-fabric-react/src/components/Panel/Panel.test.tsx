import * as React from 'react';

import * as ReactDOM from 'react-dom';

import { PanelBase } from './Panel.base';
import { PanelType } from './Panel.types';
import { Panel } from './Panel';
import * as renderer from 'react-test-renderer';
let div: HTMLElement;

// Mock Layer since it otherwise shows nothing in snapshot tests
jest.mock('../../Layer', () => {
  return {
    Layer: jest.fn().mockImplementation(props => {
      return props.children;
    })
  };
});

describe('Panel', () => {
  it('Renders Panel correctly', () => {
    const component = renderer.create(
      <Panel isOpen={true} type={PanelType.large} headerText="Large Panel">
        <span>Content goes here.</span>
        />
      </Panel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  beforeEach(() => {
    div = document.createElement('div');
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
  });

  it('fires the correct events when closing', done => {
    let dismissedCalled = false;
    let dismissCalled = false;
    const setDismissTrue = (): void => {
      dismissCalled = true;
    };
    const setDismissedTrue = (): void => {
      dismissedCalled = true;
    };

    const panel: PanelBase = ReactDOM.render(
      <PanelBase isOpen={true} onDismiss={setDismissTrue} onDismissed={setDismissedTrue} />,
      div
    ) as any;

    panel.dismiss();

    expect(dismissCalled).toEqual(true);
    expect(dismissedCalled).toEqual(false);

    setTimeout(() => {
      try {
        expect(dismissedCalled).toEqual(true);
        done();
      } catch (e) {
        done(e);
      }
    }, 250);
  });
});
