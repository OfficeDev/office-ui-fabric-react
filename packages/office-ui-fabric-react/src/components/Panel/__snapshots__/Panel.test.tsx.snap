// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Renders Panel correctly 1`] = `
<span
  className="ms-Layer"
/>
`;
